lib/Backend/Lower.cpp

```diff
@@ -1863,8 +1863,19 @@ Lowerer::LowerRange(IR::Instr *instrStart, IR::Instr *instrEnd, bool defaultDoFa
                 Assert(UNREACHED);
             }
             break;
-
+        case Js::OpCode::OverflowCheck:
+            instr->m_opcode = Js::OpCode::MOV;
+            break;
         case Js::OpCode::Conv_Prim:
+        {
+
+            IR::RegOpnd* src1 = instr->GetSrc1()->IsRegOpnd() ? instr->GetSrc1()->AsRegOpnd() : nullptr;
+            if (src1 && src1->m_sym->IsSingleDef() && src1->m_sym->GetInstrDef()->m_opcode == Js::OpCode::OverflowCheck)
+            {
+                GenerateTruncWithCheck(instr);
+                break;
+            }
+
             if (instr->GetDst()->IsFloat())
```

Comment:
Wonder if we might as well factor out instr->GetDst() and instr->GetSrc1() since they are everyone here.